test/units/module_utils/basic/test_log.py

```diff
@@ -0,0 +1,227 @@
+
+        self.am.no_log = no_log
+
+    def test_output_matches(self):
+        if sys.version_info >= (3,):
+            for msg, param in self.py3_output_data.items():
+                with patch('syslog.syslog', autospec=True) as mock_func:
+                    self.am.log(msg)
+                    mock_func.assert_called_once_with(syslog.LOG_INFO, param)
+        else:
+            for msg, param in self.py2_output_data.items():
+                with patch('syslog.syslog', autospec=True) as mock_func:
+                    self.am.log(msg)
+                    mock_func.assert_called_once_with(syslog.LOG_INFO, param)
+
+
+class TestAnsibleModuleLogJournal(unittest.TestCase):
+    """Test the AnsibleModule Log Method"""
+
+    output_data = {
+            u'Text string': u'Text string',
+            u'Toshio くらとみ non-ascii test': u'Toshio くらとみ non-ascii test',
+            b'Byte string': u'Byte string',
+            u'Toshio くらとみ non-ascii test'.encode('utf-8'): u'Toshio くらとみ non-ascii test',
+            b'non-utf8 :\xff: test': b'non-utf8 :\xff: test'.decode('utf-8', 'replace')
+            }
+
+    def setUp(self):
+        self.complex_args_token = basic.MODULE_COMPLEX_ARGS
+        basic.MODULE_COMPLEX_ARGS = '{}'
+        self.am = basic.AnsibleModule(
+            argument_spec = dict(),
+        )
+
+        self.has_journal = basic.has_journal
+        basic.has_journal = True
+        self.module_patcher = None
+
+        # In case systemd-python is not installed
+        if not self.has_journal:
+            self.module_patcher = patch.dict('sys.modules', {'systemd': MagicMock(), 'systemd.journal': MagicMock()})
+            self.module_patcher.start()
+            try:
+                reload(basic)
+            except NameError:
+                self._fake_out_reload(basic)
+
+    def tearDown(self):
+        basic.MODULE_COMPLEX_ARGS = self.complex_args_token
+        basic.has_journal = self.has_journal
+        if self.module_patcher:
+            self.module_patcher.stop()
+            try:
+                reload(basic)
+            except NameError:
+                self._fake_out_reload(basic)
+
+    def _fake_out_reload(self, module):
+        """ Something about reload being a builtin is playing havoc here.  If
+        I import it in the same method as I call it then it bashes the builtin
+        reload.  *Even if the import comes after the call*.  Since we need to
+        import it on python3, doing it in a separate method seems to make it work."""
+        from importlib import reload
+        reload(module)
+
+    @patch('systemd.journal.send')
+    def test_no_log(self, mock_func):
+        no_log = self.am.no_log
+
+        self.am.no_log = True
+        self.am.log('unittest no_log')
+        self.assertFalse(mock_func.called)
+
+        self.am.no_log = False
+        self.am.log('unittest no_log')
+        mock_func.assert_called_once()
```

Comment:
There's no [Mock](https://docs.python.org/3/library/unittest.mock.html#the-mock-class).assert_called_once().  Mock will raise an exception since version 1.3 (or Python 3.5 for the stdlib version, see  http://bugs.python.org/issue21238).  Older versions of mock will assume you're calling a method of the mocked object and will silently do nothing.

The correct check is `assert mock_func.call_count == 1`, IIRC.